lib/semantics/resolve-names.cc

```diff
@@ -218,14 +218,13 @@ class MessageHandler {
 
 private:
   // Where messages are emitted:
-  parser::Messages messages_;
+  parser::Messages *messages_;
```

Comment:
Maybe this should become a reference that's established during construction.

Reply:
I would have preferred that, but it requires passing it up through a series of superclass constructors and I decided it wasn't worth it. I'll take another look.

Reply:
In principle, it's just a fancy container. The container doesn't need to be threaded everywhere, if the contents in the container are ultimately forwarded into the "big container" owned by Semantics.